test/boost/row_cache_test.cc

```diff
@@ -3448,6 +3448,16 @@ SEASTAR_TEST_CASE(test_concurrent_reads_and_eviction) {
             auto m2 = gen();
             m2.partition().make_fully_continuous();
 
+            bool upgrade_schema = tests::random::get_bool();
+            if (upgrade_schema) {
+                schema_ptr new_schema = schema_builder(s)
+                    .with_column(to_bytes("_phantom"), byte_type)
+                    .remove_column("_phantom")
+                    .build();
+                m2.upgrade(new_schema);
+                cache.set_schema(new_schema);
+            }
+
```

Comment:
Regarding tests, I think we need a stresser that builds a model of the data outside the row cache and compares it with the real row_cache under a mix of writes, reads, schema changes, and evictions. The snapshot_source that backs the cache can read the model, no need for real sstables.